### ISSUE TITLE
Support `@deprecated` on overloaded property setters

### ISSUE THREAD

**Issue by Viicos** [python/mypy#21593]
Code sample in [Mypy playground](https://mypy-play.net/?mypy=latest&python=3.14&enable-error-code=deprecated&gist=8973ab7850bd19435367ca45b062b505)

```python
from typing import overload
from warnings import deprecated

class A:
    @property
    def t1(self) -> int:
        ...

    @t1.setter
    @deprecated("Setting t is deprecated")
    def t1(self, value: int) -> None:
        ...

    @property
    def t2(self) -> int:
        ...

    @t2.setter
    @overload
    @deprecated("Setting t to None is deprecated")
    def t2(self, value: None) -> None:
        ...

    @t2.setter
    @overload
    def t2(self, value: int) -> None:
        ...

    @t2.setter
    def t2(self, value: int | None) -> None:
        ...

a = A()

a.t1 = 1  # reported
a.t2 = None  # not reported
```

See also: https://discuss.python.org/t/107604
